core/src/main/scala/com/softwaremill/events/EventMachine.scala

```diff
@@ -45,6 +47,51 @@ class EventMachine(
       })
   }
 
+  /**
+    * Recovers db state from event log.
+    * @param timeLimit Optional point in time till which events should be recovered.
+    * @param logResult Database query result logger.
+    * @return
+    */
+  def failOverFromStoredEventsTo(timeLimit: OffsetDateTime = clock.instant().atOffset(ZoneOffset.UTC),
+    logResult: PartialFunction[Try[Unit], Unit] = recoveredActionLogger): Future[Int] = {
+    val storedEvents: DatabasePublisher[StoredEvent] = database.db.stream(eventStore.getAll(timeLimit))
+
+    val eventActions = storedEvents.mapResult(e => e.toEvent(registry.getEventClass(e.eventType)))
+      .mapResult(lookupEventAction)
+
+    val countFuture = storedEventsCountFuture().map(storedEventsCount => logger.info(s"Number of events to recover: $storedEventsCount"))
+
+    import scala.concurrent.Promise
+    val p = Promise[Int]
```

Comment:
the completed future from promise triggered actorsystem shutdown in RecoverDbState object ;) but yes, I've throw it away